src/app/campaign/store/actions/campaign-action.service.ts

```diff
@@ -45,41 +46,78 @@ export class CampaignActionService implements OnDestroy {
     if (dateRangeValid && availabilityParamsChanged) {
       this.campaignStoreService.loadAvailability(formFlight);
       if (formFlight.totalGoal) {
-        this.campaignStoreService.loadAllocationPreview(formFlight, dailyMinimum);
+        this.loadAllocationPreview(
+          formFlight.id,
+          localFlight.createdAt,
+          formFlight.set_inventory_uri,
+          formFlight.name,
+          formFlight.startAt,
+          formFlight.endAt,
+          formFlight.totalGoal,
+          dailyMinimum,
+          formFlight.zones
+        );
       }
     }
   }
 
   loadCampaignOptions() {
-    this.store.dispatch(new actions.CampaignLoadOptions());
+    this.store.dispatch(new campaignActions.CampaignLoadOptions());
   }
 
   newCampaign() {
-    this.store.dispatch(new actions.CampaignNew());
+    this.store.dispatch(new campaignActions.CampaignNew());
   }
 
   loadCampaign(id: number) {
-    this.store.dispatch(new actions.CampaignLoad({ id }));
+    this.store.dispatch(new campaignActions.CampaignLoad({ id }));
+  }
+
+  loadAllocationPreview(
+    flightId: number,
+    createdAt: Date,
+    // tslint:disable-next-line: variable-name
+    set_inventory_uri: string,
+    name: string,
+    startAt: Date,
+    endAt: Date,
+    totalGoal: number,
+    dailyMinimum: number,
+    zones: string[]
+  ) {
```

Comment:
I had the same thought, and then I thought do I really need this extra wrapper function anyway. It makes testing a bit more straightforward to have these wrapper functions rather than calling store.dispatch directly, but I might still get rid of the ones that are just wrappers around dispatch + action creator. I'm still having thoughts on this part. It's not a standard for NgRx and feels a bit extra. The service is necessary, or something that can run and watch those state changes for loading data, and better here than in components, but the rest feels like extra boilerplate.